src/strawchemy/schema/mutation/input.py

```
@@ -129,27 +126,85 @@ def from_unbound(cls, unbound: _UnboundRelationInput, model: DeclarativeBase) ->
+
+    def _register(self, attribute: MapperProperty[Any], relation_type: RelationType) -> None:
+        if attribute in self._registry:
+            return
+        if relation_type is RelationType.TO_ONE:
+            event.listens_for(attribute, "set")(self._dispatch_set)
+        else:
+            event.listens_for(attribute, "append")(self._dispatch_append)
+            event.listens_for(attribute, "remove")(self._dispatch_remove)
+        self._registry.add(attribute)
+
+    def _get_input(self, target: DeclarativeBase, initiator: Any) -> RelationInput | None:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `initiator`

(ANN401)